### PR TITLE
tests: replace nfs support test with generalized cifs test

### PR DESCRIPTION
**Pull request by zyga**
The cifs test was created relatively recently to test how a single home directory is mounted using CIFS and how a user using that home directory can use snaps. We had an earlier test for NFS with some more fragile and complex setup.

Given that NFS test fails periodically and it was broken for a while on some systems, replace the old NFS test with a variant of the CIFS test. Most of the existing test is kept intact. The portion that manages the user account is moved to the front to be easily shared with the conditional bits below.

The patch series contains a new toy executable called `okfail` which indicates that a command is allowed to fail without consequences. It should be moved to the test helpers project but it can be done on the next sync.

Jira: https://warthogs.atlassian.net/browse/SNAPDENG-31425

### CODE REVIEW

**Review by bboozzoo — APPROVED** [review]
LGTM

File: tests/main/remote-home/task.yaml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            tests.cleanup defer okfail umount /proc/fs/nfsd
+        fi
+        # If /var/lib/nfs/rpc_pipefs is not initially mounted then ask the test
+        # to unmount it later.
+        if not mountinfo.query /var/lib/nfs/rpc_pipefs .fs_type=rpc_pipefs; then
+            tests.cleanup defer okfail umount /proc/nfs/rpc_pipefs
+        fi
+
+        tests.pkgs install nfs-kernel-server
+        tests.cleanup defer tests.pkgs remove nfs-kernel-server
```
Comment (sergiocazzolato) [inline tests/main/remote-home/task.yaml:71]: it shouldn't be needed as it is done automatically

File: tests/main/remote-home/task.yaml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        retry exportfs -r
+
+        # Later on remove the exports file and reload exported filesystems.
+        tests.cleanup defer exportfs -r
+        tests.cleanup defer rm -f /etc/exports.d/test.exports
+        ;;
+      cifs)
+        # Install samba.
+        tests.pkgs install samba
+        tests.cleanup defer tests.pkgs remove samba
```
Comment (sergiocazzolato) [inline tests/main/remote-home/task.yaml:85]: same about removing pkgs is done automatically

File: tests/main/remote-home/task.yaml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+        # Create the var-home share.
+        tests.backup prepare /etc/samba
+        tests.cleanup defer tests.backup restore /etc/samba
+        # NOTE: due to <<- and the fact that this is shell-in-yaml, the spaces
+        # are followed by tabs.
+        cat <<-__SMB__ >>'/etc/samba/smb.conf'
```
Comment (sergiocazzolato) [inline tests/main/remote-home/task.yaml:92]: is it removed with samba pkg right?

**Review by sergiocazzolato — APPROVED** [review]
Thanks, just minor comments inline

File: tests/main/remote-home/task.yaml (hunk 3 above)
Comment (zyga) [reply tests/main/remote-home/task.yaml:92]: I think it is only removed on purge.

**Pull request merged**